lib/query-interface.js

```diff
@@ -193,8 +193,16 @@ QueryInterface.prototype.dropTable = function(tableName, options) {
   options = options || {};
   options.cascade = options.cascade || options.force || false;
 
-  var sql = this.QueryGenerator.dropTableQuery(tableName, options)
-    , self = this;
+  var self = this;
+
+  if (!tableName.schema && options.schema) {
```

Comment:
This code is duplicated all over this file. (it's in createTable twice....) probably a candidate for extraction into a method.